packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -280,17 +278,28 @@ const getAccountInfo = async (request: Request<MessageTypes.GetAccountInfo>) =>
             .filter((tx): tx is Transaction => !!tx);
     };
 
-    const tokenAccounts = await api.rpc
-        .getTokenAccountsByOwner(
-            publicKey,
-            { programId: address(TOKEN_PROGRAM_PUBLIC_KEY) } /* filter */,
-            {
-                encoding: 'jsonParsed',
-            },
+    const getTokenAccounts = (tokenProgramId: string) =>
+        api.rpc
+            .getTokenAccountsByOwner(
+                publicKey,
+                { programId: address(tokenProgramId) } /* filter */,
+                {
+                    encoding: 'jsonParsed',
+                },
+            )
+            .send();
+
+    const allTokenAccounts = (
```

Comment:
nit: `tokenAccounts` would be fine IMO. `allTokenAccounts` makes me think that the distinction is important and that at some point we'll filter the accounts but that isn't the case here.